walkthroughs/ec2/ec2-cluster.yaml

```diff
@@ -0,0 +1,177 @@
+Description: >
+  This template deploys an EC2 instance to the provided VPC and subnets
+
+Parameters:
+
+  EnvironmentName:
+    Description: An environment name that will be prefixed to resource names
+    Type: String
+
+  InstanceType:
+    Description: Which instance type should we use to build the EC2 cluster?
+    Type: String
+    Default: c4.large
+
+  EC2AMI:
+    Description: The Amazon Machine Image ID used for the cluster, leave it as the default value to get the latest AMI
+    Type: AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>
+    Default: /aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2
+#    Default: /aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id
+#    Default: ami-08e689a423405d7f6 amzn2-ami-hvm-2.0.20190313-x86_64-ebs
+
+  KeyName:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: The name of an existing EC2 KeyPair that will be used to access the underlying hosts in the cluster. If none appear here you need to create and download an SSH key.
+
+  ProxyRouterManagerImage:
+    Type: String
+    Description: Sidecar router manager that sets up networking for transparent proxy
+    Default: "subfuzion/aws-appmesh-proxy-route-manager:latest"
```

Comment:
we shouldn't hardcode against your version of it. I would remove the default or switch it to the ECR image (if that works)

Reply:
Ah, the ECR image doesn't work (I believe) because the instance doesn't have iam perms for ECR. you should be able to create a role with "AmazonEC2ContainerRegistryReadOnly" and then log into ecr before pulling.

Reply:
Please see the [other related comment](https://github.com/aws/aws-app-mesh-examples/pull/102#discussion_r293612042). I think open sourcing the image creation (or refactoring to use ECR vs Docker Hub) are both good things to do, but could be done as enhancements without changing the value of making this demo available now.